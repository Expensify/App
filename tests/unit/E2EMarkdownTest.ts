import {compareResults} from '../e2e/compare/compare';
import {buildMarkdown} from '../e2e/compare/output/markdown';

const results = {
    main: {
        commentLinking: [100.5145680010319, 121.8861090019345, 112.0048420019448, 124.26110899820924, 135.1571460030973, 140.33837900310755, 160.7034499980509, 158.5825610011816],
    },
    delta: {
        commentLinking: [361.5145680010319, 402.8861090019345, 412.0048420019448, 414.26110899820924, 425.1571460030973, 440.33837900310755, 458.7034499980509, 459.5825610011816],
    },
};

describe('markdown formatter', () => {
    it('should format significant changes properly', () => {
        const data = compareResults(results.main, results.delta, {commentLinking: 'ms'});
        expect(buildMarkdown(data)).toMatchSnapshot();
    });
});
