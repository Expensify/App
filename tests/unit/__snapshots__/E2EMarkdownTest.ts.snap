// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`markdown formatter should format significant changes properly 1`] = `
"## Performance Comparison Report ðŸ“Š

### Significant Changes To Duration
| Name           | Duration                                            |
| -------------- | --------------------------------------------------- |
| commentLinking | 131.681 ms â†’ 421.806 ms (+290.125 ms, +220.3%) ðŸ”´ðŸ”´ |
<details>
<summary>Show details</summary>

| Name           | Duration                                                                                                                                                                                                                                                                                                                                                                                                                                                     |
| -------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
| commentLinking | **Baseline**<br/>Mean: 131.681 ms<br/>Stdev: 19.883 ms (15.1%)<br/>Runs: 100.5145680010319 112.0048420019448 121.8861090019345 124.26110899820924 135.1571460030973 140.33837900310755 158.5825610011816 160.7034499980509<br/><br/>**Current**<br/>Mean: 421.806 ms<br/>Stdev: 30.185 ms (7.2%)<br/>Runs: 361.5145680010319 402.8861090019345 412.0048420019448 414.26110899820924 425.1571460030973 440.33837900310755 458.7034499980509 459.5825610011816 |
</details>



### Meaningless Changes To Duration
_There are no entries_

"
`;
